docs/concepts/ksql-and-kafka-streams.rst

```diff
@@ -0,0 +1,111 @@
+| Runtime included  | Yes, the KSQL server | Applications run as standard JVM processes |
++-------------------+----------------------+--------------------------------------------+
+| Queryable state   | No                   | Yes                                        |
++-------------------+----------------------+--------------------------------------------+
+
+Use the following table to help you decide between KSQL and Kafka Streams as a
+starting point for your real-time application development. 
+
++----------------------------------------------------+------------------------------------------------------+
+| Start with KSQL when…                              | Start with Kafka Streams when…                       |
++====================================================+======================================================+
+| * New to streaming and Kafka                       | * Prefer writing and deploying JVM applications      |
+|                                                    |   like Java and Scala; for example, due to           |
+|                                                    |   people skills, tech environment                    |
+| * To quicken and broaden the adoption              | * Use case is not naturally expressible through SQL, |
+|                                                    |   for example, finite state machines                 |
+|   and value of Kafka in your organization          | * Building microservices                             |
+| * Prefer an interactive experience with UI and CLI | * Must integrate with external services, or          |
+|                                                    |   use 3rd-party libraries (but KSQL UDFs may help)   |
+| * Prefer SQL to writing code in Java or Scala      | * To customize or fine-tune a use case, for example, |
+| * Use cases include enriching data; joining        |   with the Kafka Streams Processor API:              |
+|   data sources; filtering, transforming,           |   custom join variants, or probabilistic counting at |
+|   and masking data; identifying anomalous events   |   very large scale with Count-Min Sketch             |
+| * Use case is naturally expressible by using SQL,  | * Need queryable state, which KSQL doesn't support   |
+|   with optional help from User Defined Functions   |                                                      |
+| * Want the power of Kafka Streams but you          |                                                      |
+|   aren't on the JVM: use the KSQL REST API         |                                                      |
+|   from Python, Go, C#, JavaScript, shell           |                                                      |
++----------------------------------------------------+------------------------------------------------------+
+
+KSQL is usually not a good fit for BI reports and ad-hoc querying, or queries with random access patterns.

```

Comment:
Maybe add the following:
`KSQL is usually not a good fit for BI reports and ad-hoc querying, or queries with random access patterns since it's a continuous query system on data streams.`